packages/venia-ui/lib/components/App/app.js

```diff
@@ -33,6 +36,22 @@ const App = props => {
     const { markErrorHandled, renderError, unhandledErrors } = props;
 
     const [, { addToast }] = useToasts();
+    const [{ cartId }, { setCartId }] = useCartContext();
+    const [createCart] = useMutation(CREATE_CART_MUTATION);
+
+    // On initial mount create a cart if there isn't a cartId in the store.
+    // TODO: Should this belong elsewhere? Basically this is global logic that says "any time the cartId is deleted, recreate the cart".
+    useEffect(() => {
+        async function initializeCart() {
+            const {
+                data: { createEmptyCart }
+            } = await createCart();
+            await setCartId(createEmptyCart);
+        }
+        if (!cartId) {
+            initializeCart();
```

Comment:
Now, as a side effect of not having a `cartId` we re-create one.

Ideally we _never_ call the `createCart` action as it uses REST. We want to use the `createCart` mutation instead.

In this PR you'll see places I [removed](https://github.com/magento/pwa-studio/pull/1987/files#diff-ca0eebf490b4ae87e97fbec37383872bL57) calls that would create a new cart through REST. Instead, we now fetch a new cart, set the cartId, and then fetch the details for that cart.